src/Gaffer/Spreadsheet.cpp

```diff
@@ -0,0 +1,557 @@
+	}
+	return result;
+}
+
+//////////////////////////////////////////////////////////////////////////
+// RowPlug
+//////////////////////////////////////////////////////////////////////////
+
+GAFFER_PLUG_DEFINE_TYPE( Spreadsheet::RowPlug );
+
+Spreadsheet::RowPlug::RowPlug( const std::string &name, Plug::Direction direction, unsigned flags )
+	:	ValuePlug( name, direction, flags )
+{
+	addChild( new StringPlug( "name", direction ) );
+	addChild( new BoolPlug( "enabled",direction, true ) );
+	addChild( new ValuePlug( "cells", direction ) );
+}
+
+StringPlug *Spreadsheet::RowPlug::namePlug()
+{
+	return getChild<StringPlug>( 0 );
+}
+
+const StringPlug *Spreadsheet::RowPlug::namePlug() const
+{
+	return getChild<StringPlug>( 0 );
+}
+
+BoolPlug *Spreadsheet::RowPlug::enabledPlug()
+{
+	return getChild<BoolPlug>( 1 );
+}
+
+const BoolPlug *Spreadsheet::RowPlug::enabledPlug() const
+{
+	return getChild<BoolPlug>( 1 );
+}
+
+ValuePlug *Spreadsheet::RowPlug::cellsPlug()
+{
+	return getChild<ValuePlug>( 2 );
+}
+
+const ValuePlug *Spreadsheet::RowPlug::cellsPlug() const
+{
+	return getChild<ValuePlug>( 2 );
+}
+
+Gaffer::PlugPtr Spreadsheet::RowPlug::createCounterpart( const std::string &name, Direction direction ) const
+{
+	RowPlugPtr result = new RowPlug( name, direction );
+	for( const auto &cell : CellPlug::Range( *cellsPlug() ) )
+	{
+		result->cellsPlug()->addChild( cell->createCounterpart( cell->getName(), direction ) );
+	}
+	return result;
+}
+
+//////////////////////////////////////////////////////////////////////////
+// CellPlug
+//////////////////////////////////////////////////////////////////////////
+
+GAFFER_PLUG_DEFINE_TYPE( Spreadsheet::CellPlug );
+
+Spreadsheet::CellPlug::CellPlug( const std::string &name, const Gaffer::Plug *value, Plug::Direction direction )
+	:	ValuePlug( name, direction )
+{
+	addChild( new BoolPlug( "enabled", direction, true ) );
+	addChild( value->createCounterpart( "value", direction ) );
+	valuePlug()->setFlags( Plug::Dynamic, false );
+}
+
+BoolPlug *Spreadsheet::CellPlug::enabledPlug()
+{
+	return getChild<BoolPlug>( 0 );
+}
+
+const BoolPlug *Spreadsheet::CellPlug::enabledPlug() const
+{
+	return getChild<BoolPlug>( 0 );
+}
+
+Gaffer::PlugPtr Spreadsheet::CellPlug::createCounterpart( const std::string &name, Direction direction ) const
+{
+	return new CellPlug( name, valuePlug(), direction );
+}
+
+//////////////////////////////////////////////////////////////////////////
+// Spreadsheet
+//////////////////////////////////////////////////////////////////////////
+
+size_t Spreadsheet::g_firstPlugIndex = 0;
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( Spreadsheet );
+
+Spreadsheet::Spreadsheet( const std::string &name )
+	:	ComputeNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	addChild( new BoolPlug( "enabled", Plug::In, true ) );
+	addChild( new StringPlug( "selector" ) );
+	addChild( new RowsPlug( "rows" ) );
```

Comment:
`NameSwitch` adds an empty row when its configured for any given plug type. Would it make sense to add a row to the `Spreadsheet` too?